app/Models/User.php

```diff
@@ -59,8 +60,28 @@ class User extends Authenticatable
         'email_verified_at' => 'datetime'
     ];
     
+    /**
+     * @var
+     */
+    public $products;
+
+    /**
+     * @var
+     */
+    public $cartItem;
+
     public function products()
     {
-        return $this->hasMany('App\Models\Product', 'own_by');
+        return $this->hasMany('App\Models\Product', 'id', 'own_by');
```

Comment:
this is the other way around...

return $this->hasMany('App\Models\Product', 'own_by','id');

illustartion -> return $this->belongsTo('App\Models\Product', 'foreign_key', 'other_key');

Reply:
and the only time you add 'other_key' or 'local_key' is if User has a different PKey other than ID